testing-bundle-structure.py

```diff
@@ -0,0 +1,380 @@
+
+
+
+    # TODO check compatibility with ID package 
+    def checking_id_package_compatibility(self):
+
+        try:
+            if self.compiled_em_yaml_text["transformations"]["input"] == {'source': '$sources.input', 'operations': []}:
+                self.successes["ID Package Compatibility"].append("Correct transformations.input")
+            else:
+                raise InvalidBlockError
+        except (KeyError, InvalidBlockError):
+            self.warnings["ID Package Compatibility"].append("Please make sure the following block is in your `earthmover.yaml`:\n\n"
+                                                                "transformations:\n"
+                                                                "  input:\n"
+                                                                "    source: '$sources.input'\n"
+                                                                "    operations: []")
+
+
+        try:
+            if self.compiled_em_yaml_text["config"]["parameter_defaults"]["STUDENT_ID_NAME"] == "edFi_studentUniqueID":
+                self.successes["ID Package Compatibility"].append("Correct config.parameter_defaults.STUDENT_ID_NAME")
+            else:
+                raise InvalidBlockError
+        except (KeyError, InvalidBlockError):
+            self.warnings["ID Package Compatibility"].append("Please make sure the following block is in your `earthmover.yaml`:\n\n"
+                                                                "config:\n"
+                                                                "  parameter_defaults:\n"
+                                                                '    STUDENT_ID_NAME: "edFi_studentUniqueID"')
+                
+
+        self.not_run_tests.remove("ID Package Compatibility")
+
+
+    # TODO: exit testing
+    def exit_testing(self):
+
+        """✨ ⚠️ ✅"""
+
+
+        for key in self.warnings.keys():
+            if key in self.not_run_tests:
+                continue
+
+            print(f"\n✨ {key}")
+
+            for warning in self.warnings[key]:
+                print(f"    ⚠️  {warning}")
+
+            for success in self.successes[key]:
+                print(f"    ✅ {success}")
+
+
+        if len(self.not_run_tests) > 0:
+            print("\n⚠️  Not run tests:")
+            for test in self.not_run_tests:
+                print(f"        {test}")
+
+        print()
+
+
+def main():
+    
+    if len(sys.argv) < 2:
+        print('Usage: python testing-bundle-structure.py <folder_path>')
+        sys.exit(1)
+
+    folder_path = sys.argv[1]
+                # "./assessments/{assessment_name}"
+
+    c = Checker(folder_path)
+
+    try:
+        c.checking_folder_contents()
+        c.getting_params()
+
+        try:
+            c.compiling_yaml()
+            c.parsing_compiled_em_yaml()
+            c.checking_id_package_compatibility()
+        except NoEMParams:
+            pass
```

Comment:
Since you have some branched dependencies here (tests that can only run if other tests didn't error), it might be helpful to add some additional comments at a higher level about what these dependencies are. e.g. "We can only try to compile the Earthmover config if we successfully parsed the README"